lms/djangoapps/grades/tests/test_signals.py

```diff
@@ -148,3 +158,83 @@ def test_score_reset_bad_user(self):
         self.signal_mock.assert_not_called()
+
+
+@ddt.ddt
+class ScoreChangedUpdatesSubsectionGradeTest(ModuleStoreTestCase):
+    """
+    Ensures that upon SCORE_CHANGED signals, the handler
+    initiates an update to the affected subsection grade.
+    """
+    def setUp(self):
+        super(ScoreChangedUpdatesSubsectionGradeTest, self).setUp()
+
+    def setUpCourseAndUser(self, enable_subsection_grades):
+        """
+        Configures the course for this test.
+        """
+        self.user = UserFactory()
+        self.course = CourseFactory.create(
+            org='edx',
+            name='course',
+            run='run',
+            metadata={'enable_subsection_grades_saved': enable_subsection_grades})
+        
+        self.chapter = ItemFactory.create(parent=self.course, category="chapter", display_name="Chapter")
+        self.sequential = ItemFactory.create(parent=self.chapter, category='sequential', display_name="Open Sequential")
+        self.problem = ItemFactory.create(parent=self.sequential, category='problem', display_name='problem')
+
+        self.score_changed_kwargs = {
+            'points_possible': 10,
+            'points_earned': 5,
+            'user_id': self.user.id,
+            'course_id': unicode(self.course.id),
+            'usage_id': unicode(self.problem.location),
+        }
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_subsection_grade_updated_on_signal(self, default_store):
+        with self.store.default_store(default_store):
+            self.setUpCourseAndUser(True)
+            with check_mongo_calls(2) and self.assertNumQueries(18):
+                recalculate_subsection_grade_handler(None, **self.score_changed_kwargs)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_query_count_does_not_change_with_more_problems(self, default_store):
+        with self.store.default_store(default_store):
+            self.setUpCourseAndUser(True)
+            ItemFactory.create(parent=self.sequential, category='problem', display_name='problem2')
+            ItemFactory.create(parent=self.sequential, category='problem', display_name='problem3')
+            with check_mongo_calls(2) and self.assertNumQueries(18):
+                recalculate_subsection_grade_handler(None, **self.score_changed_kwargs)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_subsection_grades_not_enabled_on_course(self, default_store):
+        with self.store.default_store(default_store):
+            self.setUpCourseAndUser(False)
+            with check_mongo_calls(2) and self.assertNumQueries(1):
+                recalculate_subsection_grade_handler(None, **self.score_changed_kwargs)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_SUBSECTION_GRADES_SAVED': False})
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_subsection_grades_not_enabled(self, default_store):
```

Comment:
If you do, you'll introduce variability from other handlers of that signal.
To be comprehensive while we're working on this (eventually, we may delete some of these permutations):
1. call score_changed.send, feature on/off, split/mongo
2. call your specific handler, feature on only, split/mongo